rspace/src/main/scala/coop/rchain/rspace/ReplayRSpace.scala

```diff
@@ -179,22 +183,21 @@ class ReplayRSpace[F[_]: Sync, C, P, A, K](
       channels =>
         store
           .getContinuations(channels)
-          .map(_.zipWithIndex.filter {
+          .map(_.iterator.zipWithIndex.filter {
             case (WaitingContinuation(_, _, _, _, source), _) =>
               comm.consume == source
-          }),
+          }.toSeq),
       c =>
         store
           .getData(c)
-          .map(_.zipWithIndex)
-          .map(
-            as =>
-              c -> {
-                if (c == channel)
-                  Seq((Datum(data, persist, produceRef), -1))
-                else as
-              }.filter { matches(comm) }
-          )
+          .map { as =>
+            val d = Datum(data, persist, produceRef)
+            val matching = if (c == channel) {
+              if (comm.matches(d, produceCounter.get(produceRef))) Seq(d -> -1) //WTF?
```

Comment:
Why the "wtf"?

Reply:
Oh, sorry, probably a left over from my workings. I'll remove that. (Well, this `-1` is just a clever hack_ for `store.removeDatum(candidateChannel, dataIndex).whenA(dataIndex >= 0 && !persistData)` to work. I don't  feel confidence in such hacks (that is, passing "special" values as kind of "wink" to other parts of system that "wink back" because they happen to understand the hack)